### DISCUSSION TITLE
Question regarding fix #3808 about triggering WhenActivated

### DISCUSSION THREAD

**Discussion by Geccoka** [reactiveui/ReactiveUI#3825] [Q&A]
Hi, i have a question regarding PR #3808 . Since that PR is closed i cannot comment there so i created this discussion instead.

The fix is only made for ReactiveInjectableComponentBase. What about ReactiveComponentBase and ReactiveLayoutComponentBase? I think the problem exists in them as well.

Right now if i have a component inherited from ReactiveComponentBase<ActivatableViewModel> i have to put at least an empty WhenActivated call in the component ensuring that the VM's WhenActivated will be called:

```
    protected override void OnInitialized()
    {
        this.WhenActivated(disposables => { });

        base.OnInitialized();
    }
```

If i don't do this the VM's WhenActivated will never be called by the framework.

The other thing, wouldn't be better to put the if statement with the subscriptions before the __initSubject.OnNext(Unit.Default);_ line?

So instead of this:

```
    protected override void OnInitialized()
    {
        _initSubject.OnNext(Unit.Default);
        if (ViewModel is IActivatableViewModel avm)
        {
            Activated.Subscribe(_ => avm.Activator.Activate()).DisposeWith(_compositeDisposable);
            Deactivated.Subscribe(_ => avm.Activator.Deactivate());
        }

        base.OnInitialized();
    }
```

like this:

```
    protected override void OnInitialized()
    {
        if (ViewModel is IActivatableViewModel avm)
        {
            Activated.Subscribe(_ => avm.Activator.Activate()).DisposeWith(_compositeDisposable);
            Deactivated.Subscribe(_ => avm.Activator.Deactivate());
        }

        _initSubject.OnNext(Unit.Default);

        base.OnInitialized();
    }
```

**Comment by ChrisPulman** [comment]
Thank you for raising this, I will check this and the other  locations where we need to fix the same issue.

**Reply by martinbu** [reply]
I think this is fixed now via #3871. Does this mean, that a call to `this.WhenActivated(disposables => { });` in the view is not neccessary anymore? If so, should the [documentation](https://www.reactiveui.net/docs/handbook/when-activated.html#views) be updated, because the following comment seems to be confusing now? 

```csharp
this.WhenActivated(disposables =>
{   
    // If you put the WhenActivated block into your IViewFor
    // implementation constructor, the view model will also
    // get activated if it implements IActivatableViewModel.
```

**Reply by ChrisPulman** [reply]
The latest update was for Blazor, I need to check the other platforms to see if an update is required for them too, once a release is made I will revise the documentation too. Thank you for highlighting this.

**Reply by ChrisPulman** [reply]
I believe that it is only Blazor that has the ability to have ViewModel Activation without having a call to `this.WhenActivated(disposables => { });` on the View.

I agree that there should be documentation that shows how the Blazor WhenActivated works, as it is different.
I have other documentation to amend too and will add documentation for Blazor too.
